freshmaker/image_verifier.py

```diff
@@ -178,26 +172,69 @@ def verify_repository(self, repo_name):
+
+        def sorted_by_nvr(lst, reverse=False):
+            """
+            Sorts the list `lst` containing NVR by the NVRs.
+
+            :param list lst: List with NVRs to sort.
+            :param bool reverse: When True, the result of sorting is reversed.
+            :rtype: list
+            :return: Sorted `lst`.
+            """
+            def _compare_items(item1, item2):
+                nvr1 = item1["brew"]["build"]
+                nvr2 = item2["brew"]["build"]
+
+                nvr1_dict = kobo.rpmlib.parse_nvr(nvr1)
+                nvr2_dict = kobo.rpmlib.parse_nvr(nvr2)
+                if nvr1_dict["name"] != nvr2_dict["name"]:
+                    return _cmp(nvr1_dict["name"], nvr2_dict["name"])
+                return kobo.rpmlib.compare_nvr(nvr1_dict, nvr2_dict)
+
+            return sorted(
+                lst, key=functools.cmp_to_key(_compare_items), reverse=reverse)
+
+        # There can be multi-arch images which share the same
+        # image['brew']['build']. Freshmaker is not interested in the image
+        # architecture, it is only interested in NVR, so group the images
+        # by the same image['brew']['build'] and include just first one in the
+        # image list.
+        sorted_images = sorted_by_nvr(images, reverse=True)
+        images = []
+
+        # We must combine content_sets with same image NVR
+        # but different architectures into one content_sets field
+        for k, temp_images_iter in groupby(sorted_images, key=lambda item: item["brew"]["build"]):
+            temp_images: List = list(temp_images_iter)
+            img = temp_images[0]
+            if 'content_sets' in img and len(temp_images) > 1:
+                new_content_sets = set(img.get('content_sets'))
+                for i in temp_images[1:]:
+                    new_content_sets.update(i.get('content_sets', []))
+                img["content_sets"] = list(new_content_sets)
+            images.append(img)
+
         for image in images:
             self._verify_image_data(image)
-            image_data = {"content_sets": [], "tags": []}
-            for repodata in image['repositories']:
+            image_data: Dict[str, List[str]] = {"content_sets": [], "tags": []}
+            for repodata in image["repositories"]:
                 if repodata["repository"] == repo_name:
                     image_data["tags"] = [t["name"] for t in repodata["tags"]]
                     break
             image_data["content_sets"] = image["content_sets"]
-            data["images"][image.nvr] = image_data
+            data["images"][image["brew"]["build"]] = image_data
 
         if not data["images"]:
             raise ValueError(
-                "No published images tagged by %r found in repository" % (
-                    repo["auto_rebuild_tags"]))
+                "No images found in repository %r"
+                % (repo["repository"])
+            )
```

Comment:
This is misleading, it doesn't mean we don't find images in this repository, instead of that, we just can't find images in that repository which are tagged with the auto-rebuild tags, so this should not be changed.